tests/emukit/experimental_design/test_batch_experimental_design.py

```diff
@@ -0,0 +1,110 @@
+import GPy
+import mock
+import numpy as np
+import pytest
+
+from emukit.core import ParameterSpace, ContinuousParameter
+
+from emukit.model_wrappers import GPyModelWrapper
+
+from emukit.core.acquisition import Acquisition
+from emukit.core.interfaces import IModel
+from emukit.core.loop.loop_state import create_loop_state
+from emukit.core.optimization import AcquisitionOptimizer
+from emukit.core.loop.candidate_point_calculators import GreedyBatchPointCalculator
+from emukit.experimental_design.model_based.experimental_design_loop import ExperimentalDesignLoop
+
+
+def test_batch_experimental_design_loop():
+
+    class MockModel(GPyModelWrapper):
+        def optimize(self):
+            # speed up test by skipping the actual hyper-parameter optimization
+            pass
+
+    user_function = lambda x: x
+
+    space = ParameterSpace([ContinuousParameter('x', 0, 3)])
+
+    # Make model
+    x_init = np.linspace(0, 3, 5)[:, None]
+    y_init = user_function(x_init)
+    gpy_model = GPy.models.GPRegression(x_init, y_init)
+    model = MockModel(gpy_model)
+
+    loop = ExperimentalDesignLoop(space, model, batch_size=5)
+    loop.run_loop(user_function, 5)
+
+    assert(loop.loop_state.iteration == 5)
+    assert(loop.loop_state.X.shape[0] == 30)
+
+
+def test_batch_point_calculator():
+    class MockModel(IModel):
```

Comment:
This mock model definition repeats twice. Let's refactor to make it a fixture. It can stay in this file for the time being though